packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -2661,18 +2663,42 @@ function updateDehydratedSuspenseComponent(
     // This boundary is in a permanent fallback state. In this case, we'll never
     // get an update and we'll never be able to hydrate the final content. Let's just try the
     // client side render instead.
+    let serverError;
+    if (__DEV__) {
+      const {
+        errorMessage,
+        errorComponentStack,
+        errorHash,
+      } = getSuspenseInstanceFallbackErrorDetails(suspenseInstance);
+      if (errorMessage) {
+        serverError = errorMessage;
+      }
+      if (errorHash) {
+        serverError =
+          (serverError ? serverError + '\nServer Error Hash: ' : '') +
+          errorHash;
+      }
+      if (serverError && errorComponentStack) {
+        serverError += errorComponentStack || '';
+      }
+    } else {
+      serverError = getSuspenseInstanceFallbackErrorHash(suspenseInstance);
+    }
     return retrySuspenseComponentWithoutHydrating(
       current,
       workInProgress,
       renderLanes,
       // TODO: The server should serialize the error message so we can log it
       // here on the client. Or, in production, a hash/id that corresponds to
       // the error.
-      new Error(
-        'The server could not finish this Suspense boundary, likely ' +
-          'due to an error during server rendering. Switched to ' +
-          'client rendering.',
-      ),
+      serverError
+        ? // eslint-disable-next-line react-internal/prod-error-codes
+          new Error(serverError)
```

Comment:
So we're going to do a follow up PR to expand the onRecoverableError API, right? Like add a componentStack?

In that API where does the hash go?

I could imagine for example that the hash goes into the errorInfo object as a second argument:

```
onRecoverableError(error, {
  componentStack: '...',
  hash: '...',
});
```

In that world the error message would just be message. So I think we just make this `new Error(errorMessage)` for parity with other errors. 

The hash could also alternatively go onto this error object itself because it's not applicable to other errors passed to `onRecoverableError` so the argument on errorInfo isn't necessarily generally applicable.

```
let error = new Error(errorMessage);
error.hash = errorHash;
```

Reply:
The default error reports (`reportError` and `console.error`) don't expose ad hoc properties in the console so attaching them to the error object will not actually expose them unless one implements onRecoverableError and knows to look for them

if we pass a second arg to reportError it is ignored, but if we pass a second arg to console.error (through the defaultOnRecoverableError implementation) we would get a visible rendition of the errorInfo but we would need to stop using reportError as the default handler.

Do you expect frameworks to implement a baseline defualt onRecoverableError handler that knows how to unwrap and utilize these extra properties so that the lack of default exposure is not a concern?

As for using the errorMessage, are you suggesting we use the "there was an error on the server, switched to client rendering" message that we fall back to if no message is provided (in Dev) in prod 100% of the time?

Reply:
> Do you expect frameworks to implement a baseline defualt onRecoverableError handler that knows how to unwrap and utilize these extra properties so that the lack of default exposure is not a concern?

Yes in particular for the hash. For one you wouldn't have a hash in the first place unless you also implemented `onError` on the server and the hash is useless to someone that doesn't have a storage for the value that they can look up. You'd have to know to do this. I expect a richer environment to just do that automatically translate the hash to a real error message in the UI for prod.

> As for using the errorMessage, are you suggesting we use the "there was an error on the server, switched to client rendering" message that we fall back to if no message is provided (in Dev) in prod 100% of the time?

Yea and in DEV it would be the original message on the server.

Reply:
>Perhaps as a convenience we can also include the hash only in the prod error message similar to what we do with the minified error message.

ah like 

"`The server could not finish this Suspense boundary, likely due to an error during server rendering. Switched to client rendering. (Server Error Hash: ${theHash})`"

Reply:
Yea. I didn’t want to mess with the error message in dev since it’s user provided but for our own it’s fine.